### DISCUSSION TITLE
How to run old Direct Draw games on M1?

### DISCUSSION THREAD

**Discussion by nevack** [The-Wineskin-Project/WineskinServer#364] [Q&A]
The game I tried to run is [Harry Potter and the Chamber of Secrets](https://www.pcgamingwiki.com/wiki/Harry_Potter_and_the_Chamber_of_Secrets)
The game is made with UE1 and is 32bit, obviously.

Wine is used via Wineskin wrapper with WS11 engines.

The main problem I've got is broken launcher - menu isn't rendered properly.
Linux users can fix this by setting virtual desktop resolution to some low value, like 1024x768.
This option is greyed out in macOS `winecfg`.

I managed to run game by providing some quick-loading (`PrivetDr.unr -LOADGAME=0 -SAVESLOT=1`).

Everything renders good, but performance is really bad.
What I have tried:
1. dgVoodoo2 and `ddraw` dll override.
2. [DirectX9 renderer](https://www.moddb.com/mods/directx-9-renderer-for-stock-hp2)
3. CNC DDraw and `ddraw` dll override.

I haven't tried DXVK, as I thought DXVK works only with 64bit apps.
I had a hope CX22.1.0 would fix this (CodeWeavers says `Support for 32-bit DirectX 10/11 games.` on macOS), but the issue persists.

Is there some general steps or recommendations to fix or at least debug what is going wrong?

CCing @Gcenx for help, it's obviously not that urgent, but you expertise will be much appreciated.

**Comment by Gcenx** [comment]
From the little I’ve managed to dig up about this game it uses DirectX7 as it’s rendering backend. If the game also provides a ddraw backend you’d need to select that and install the `cnc_ddraw` verb in winetricks.

I’ve never used dgVoodoo2 but the setup should be basically the same as cnc-ddraw. Make sure the game is using the ddraw backend and add a livery override for `*ddraw` and set to `native,builtin`

The DirectX9 wrapper won’t be useful as that still runs on-top of OpenGL and Apple silicon doesn’t have native OpenGL just a basic compatibly wrapper running on-top of Metal for WebGL.

DXVK verbs currently support 64Bit DirectX10/11, a later update to winetricks I’ll be supporting 32Bit DirectX10/11 on macOS Catalina and later. Still this won’t be useful for this game as it doesn’t run on DirectX10/11.

**Reply by nevack** [reply]
Thanks for your reply!

By switching to `ddraw` you mean switching to Software Rendering?

Also there's option 2 - use DirectX 9 backend.
Is there translation wrapper for DirectX 9?

**Reply by Gcenx** [reply]
There’s no option called `ddraw` it would be named `Direct Draw`

Software rendering means everything is done on the CPU instead of the usual of GPU rendering.

And no DirectX9 isn’t an option as that will still be rendered by WineD3D over OpenGL and that’s slow AF especially on Apple Silicon.

They’re might be other options available that I’m not  aware of.

**Reply by kode54** [reply]
Why isn't Direct3D 9 using DXVK? Is MoltenVK so incomplete that it can't support DirectX 9?
